### DISCUSSION TITLE
How about dispatching core_app_run_after event before session close?

### DISCUSSION THREAD

**Discussion by hirale** [OpenMage/magento-lts#4031] [Ideas]
It's more convenient to edit session in core_app_run_after event.  What do you think?

![image](https://github.com/OpenMage/magento-lts/assets/22028058/71c12400-acea-4284-9b33-ef30e3f6e6a5)

**Comment by kiatng** [comment]
I am curious on the use case, can you elaborate?

**Comment by hirale** [comment]
When performing asynchronous operations, data required for these operations can be temporarily stored in the session using other events. Then, the data can be retrieved in the `core_app_run_after` event, processed, and subsequently removed from the session.

For example, if you want to send information about an "add to wishlist" action to an analytics provider.
You can use the `wishlist_product_add_after` event to capture the item information and save it in the session. 
Later, in the `core_app_run_after` event, you can handle the relevant logic for sending the data. 
Finally, after processing, the item information should be removed from the session.

**Comment by kiatng** [comment]
Have you look at the original PR #1592, which is very specific on closing the session. Your proposed change would modify that.

**Comment by hirale** [comment]
I read it, but there are any better solutions for this type of scenarios? It must be save to db or a queue?

**Comment by hirale** [comment]
Finally I have created a simple queue module to do this.

[openmage-redis-queue](https://github.com/hirale/openmage-redis-queue)

**Comment by luigifab** [comment]
You can simply store the data in any model (not in session) with a destruct method that perform the sync.

With the mentioned PR, destruct methods are called after response flushes, so there is no incidence for the client.
